### ISSUE TITLE
NAV-GAC-Abonos de venta - check de factura electrónica

### ISSUE THREAD

**Issue by bpesquero** [PKFEric/webProxima#1082]
Estoy observando que los abonos parece que no arrastran bien el campo de factura eletrónica. Esto está provocando que la impresión y envío de email no sea correcta.
Pongo de ejemplo el siguientes

AC/202100027--> está marcado como factura electrónica NO. El contrato tiene el check y las dos unicas facturas de venta que tienen también tienen el check correcto.

**Comment by JMunoz17** [comment]
El abono de venta AC/2021000027 correspondiente al contrato COM/03504 tiene marcado a SI el campo factura electrónica.

Estáis teniendo algún problema con la impresión o el envío del email? O lo has dicho porque has visto el check marcado a NO? He visto que la lista de abonos a la que se accede desde la ficha del contrato tenia mal sacado este campo por lo que siempre ponía NO en Factura Electrónica.

Ya tenéis esa lista arreglada y la realidad es que estaba abono si esta marcado como factura electrónica por lo que impresión y envío de mail deberían ser correctos. Confirmadme que así es.

**Comment by bpesquero** [comment]
Efectivamente, lo que estuve viendo durante el fin de semana es que el problema realmente estaba en la lista. No sólo con el campo de factura electrónica, sino también con el campo de cod lote. ¿habéis revisado ese campo?
Gracias

**Comment by JMunoz17** [comment]
El código de lote es la dimensión 5 y por lo que veo el dato que se muestra es el correcto. Por que dices que no esta bien?

**Comment by bpesquero** [comment]
Ejemplo AC/202100025--> tiene codigo lote EMAIL, o al menos el contrato lo tiene desde hace tiempo, y en la lista aparece como vacío

**Comment by JMunoz17** [comment]
En los abonos las dimensiones vienen de la factura que se esta corrigiendo y no del contrato. Esta bien, la factura corregida correspondiente no tiene el cód. lote email. En la lista se muestra el campo de forma correcta.

Aquí puedes ver que el ultimo abono si que tiene el cód. lote porque corrige una factura que ya lo tenia.

![image](https://user-images.githubusercontent.com/26869276/109483659-d05b1f00-7a7f-11eb-8828-cdf4b7f39f35.png)

**Comment by bpesquero** [comment]
No sé si hay opción, pero realmente esta dimensión no debería heredarse de la factura a la que abono sino que debería traerse del contrato
Gracias

**Comment by JMunoz17** [comment]
Pues como nos digas.. poderse se puede hacer.
Ahora mismo en los abonos todas las dimensiones vienen de la factura corregida, habría que determinar que hacemos con cada una de las dimensiones.

Todo depende del uso que les deis.. para que usáis la de Cód. Lote?

Lo normal es que cuando haces un abono de una factura ambas tengan las mismas dimensiones..

**Comment by bpesquero** [comment]
El código de lote se usa, combinado con el check de factura electrónica, para determinar si el documento (abono) se envía por emial, por correo ordinario, se sube a la web,... Es por ello, que para que tenga sentido, se tiene que traer del contrato, igual que el check de factura electrónica, ya que si no, nos encontramos con cosas incongruentes.

**Comment by JMunoz17** [comment]
El campo de factura electrónica se valida al registrar tanto facturas como abonos y se toma el valor del contrato, por lo que con ese no hay ningún problema.

Para el código de lote si es eso lo que necesitáis podemos modificar el proceso de facturación para que al crear un abono lo tome del contrato en lugar de la factura corregida.

Seria necesario actualizar este valor al registrar el documento? O con hacerlo al generarlo es suficiente?

Solo lo hacemos para la dimensión 5 cod. lote no?

**Comment by bpesquero** [comment]
Perdón por la demora. Si sólo con la dimensión 5 y es suficiente con hacerlo en la genearción del documento.

**Comment by JMunoz17** [comment]
Subido a PRE y PROD.
